boringtun/src/noise/timers.rs

```diff
@@ -385,10 +378,9 @@ impl Tunn {
     pub fn time_since_last_handshake_at(&self, now: Instant) -> Option<Duration> {
         let current_session = self.current;
         if self.sessions[current_session % super::N_SESSIONS].is_some() {
-            let duration_since_tun_start = now.duration_since(self.timers.time_started);
-            let duration_since_session_established = self.timers[TimeSessionEstablished];
+            let session_established_at = self.timers[TimeSessionEstablished];
 
-            Some(duration_since_tun_start - duration_since_session_established)
+            Some(now.duration_since(session_established_at))
```

Comment:
Is this logic correct?

The original code is calculating the time since last handshake as the difference between the tunnel start and session established time.

The new code calculates it as the difference between the now and the session established time.

These quantities are different, no?

Reply:
All timers in boringtun are currenty `Duration`s and defined as "seconds since tunnel started" (that is what is being changed with this PR) even though semantically `TimeSessionEstablished` is a point-in-time.

So in order to find out, how much time has passed since the last handshake, I need to first calculate, how much time has (relative to `now`) has passed since the tunnel got started. If I then substract that time from the other duration, I get the actual duration relative to `now`.

The prior code is super confusing (but has been working in production). The new code is a lot easier to read here so I think this is correct.

Reply:
Ah, I see. So the naming in the old correct is what's incorrect.